### PR TITLE
UAF-259 EDS Modification for UAIdentityManagementPersonDocument changed from OJB to JPA

### PR DESCRIPTION
**Pull request by amandazhangm**
1. Moved UA IdentityMangementPersonDocument cutomized Java and xml from kim-impl and kim-ldap to rice-impl to align with vanilla IdentityMangementPersonDocument directory and jar build.

2. Since OJB is no longer used for rice-impl by rice, got to redefine UAIdentityManagementPersonDocument entity inheritance with EclipseLink JPA with details below. 

- Added JPA annotation to UaIdentityManagementPersonDocument.java with entity and table mapping.

- Defined PersonDocumentExtractor to annotate IdentityManagementPersonDocument to identify which entity class should be used to map table entry

- Defined IdentityManagementPersonDocumentCustomizer and UAIdentityManagementPersonDocumentCustomizer to define different customized query to KRIM_PERSON_DOCUMENT_T as it's mapping to two different entity class, one vanilla and one UA class.

-  Added UaIdentityManagementPersonDocument.java to Spring JPA path to scan

3. Modified IdentityManagementPersonDocument workflow xml with removing attributes tag which had reference to KFS searching attributes but should not be used for RICE

4.  Refactoring edu new class comments

### CODE REVIEW

File: rice-middleware/impl/src/main/java/edu/arizona/rice/kim/service/UaUiDocumentService.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * Copyright 2009 State of Arizona Board of Regents
  *
  */
-package edu.arizona.kim.service.impl;
+package edu.arizona.rice.kim.service;
```
Comment (nataliac06) [inline rice-middleware/impl/src/main/java/edu/arizona/rice/kim/service/UaUiDocumentService.java:6]: This is a Service Implementation and I think it should be refactored 
1. moved to ...edu/arizona/rice/kim/service/impl
2. renamed to UaUiDocumentServiceImpl.java

**Review by nataliac06 — CHANGES_REQUESTED** [review]
Looks great, just one quirk...

File: rice-middleware/impl/src/main/java/edu/arizona/rice/kim/service/UaUiDocumentService.java (hunk 1 above)
Comment (amandazhangm) [reply rice-middleware/impl/src/main/java/edu/arizona/rice/kim/service/UaUiDocumentService.java:6]: Thank you for catching this, Natalia. This code was migrated from rice-2.1-ua-development originally that I didn't look into its package and name. I made recommended code change and will retest, commit and squash.

**Comment by amandazhangm** [comment]
Close and will create new PR for code review changes.

**Pull request closed without merging**